cmd/conduit/list.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/algorand/indexer/exporters"
+	"github.com/algorand/indexer/importers"
+	"github.com/algorand/indexer/processors"
+)
+
+func makeListCmd() *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "lists all plugins available to conduit",
+		Long:  "lists all plugins available to conduit",
+		Args:  cobra.MaximumNArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runListCmd(os.Stdout, args)
+		},
+		SilenceUsage: true,
+		// Silence errors because our logger will catch and print any errors
+		SilenceErrors: true,
+	}
+
+	return cmd
+}
+
+func runListCmd(out io.Writer, listArgs []string) error {
+	if len(listArgs) == 0 {
+		output, err := outputAll()
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprint(out, output)
+		return nil
+	}
+
+	if len(listArgs) == 1 {
+		return fmt.Errorf("invalid argument, specify [importers, processors, exporters] before plugin name")
+	}
+
+	pluginTypeSpecifier := listArgs[0]
```

Comment:
Can we downcase and remove space from this so that `conduit list  Importers   foobar` still works as expected. It's possible that cobra is already removing spaces.